app/controllers/payment_controller.rb

```diff
@@ -1,43 +1,84 @@
 # frozen_string_literal: true
+
 class PaymentController < ApplicationController
   skip_before_action :verify_authenticity_token
 
   def transaction
-    builder = if recurring?
-                client::Subscription.make_subscription(payment_options)
-              else
-                client::Transaction.make_transaction(payment_options)
-              end
-
     if builder.success?
-      write_member_cookie(builder.action.member_id) unless builder.action.blank?
-      id = recurring? ? { subscription_id: builder.subscription_id } : { transaction_id: builder.transaction_id }
-
-      respond_to do |format|
-        format.html { redirect_to follow_up_page_path(page) }
-        format.json { render json: { success: true }.merge(id) }
-      end
+      process_and_render_success
```

Comment:
This is definitely a style improvement, but I would say this would be better if instead it called each piece of the success from here rather than called a function to do it, ie
```
write_member_cookie(builder.action.member_id) unless builder.action.blank?
store_in_vault if store_in_vault?
update_on_ak(builder.action.member_id)
render_success
```